ppb/features/default_sprites.py

```diff
@@ -58,3 +60,118 @@ def _exponential_decay(self, time_delta):
         remaining = decay_rate ** time_delta
         decay_amount = 1. - remaining
         return decay_amount
+
+
+@dataclass
+class DirectionKeys:
+    up: ppb.keycodes
+    left: ppb.keycodes
+    down: ppb.keycodes
+    right: ppb.keycodes
+
+
+@dataclass
+class ArrowDirectionKeys(DirectionKeys):
+    up: ppb.keycodes = ppb.keycodes.Up
+    left: ppb.keycodes = ppb.keycodes.Left
+    down: ppb.keycodes = ppb.keycodes.Down
+    right: ppb.keycodes = ppb.keycodes.Right
+
+
+@dataclass
+class WASDDirectionKeys(DirectionKeys):
+    up: ppb.keycodes = ppb.keycodes.W
+    left: ppb.keycodes = ppb.keycodes.A
+    down: ppb.keycodes = ppb.keycodes.S
+    right: ppb.keycodes = ppb.keycodes.D
```

Comment:
I'm trying to decide how I feel about using dataclasses for this--I'm feeling like not, since it's an additional thing to import.

Reply:
no worries, I'll remove this dependency

Reply:
That doesn't actually help. The problem wasn't the use of specifically `dataclasses` (which are in the stdlib and used elsewhere in the code). The problem is the ergonomics of using this kind of "object as a bag" in this context. In addition, by removing the use of `dataclass`, you have made it less usable by no longer having an `__init__`.

The problem is that in order to customize this, you have to import an additional class, for something that we generally accomplish with a dictionary:

```python
{
    keycodes.W: Vector(-1, 0),
    keycodes.Left: Vector(-1, 0),
    keycodes.D: Vector(-1, 0),
    keycodes.Right: Vector(-1, 0),
}
```

Which has two benefits:
* Less boilerplate
* More flexibility

Reply:
To summarize: just because you're not using `dataclasses` doesn't mean you're not writing data classes.

Reply:
I see what u mean now, I've used a dictionary instead